spotbugs/src/main/java/edu/umd/cs/findbugs/ba/vna/ValueNumberFrame.java

```diff
@@ -54,6 +54,8 @@
 
     private Map<AvailableLoad, ValueNumber[]> availableLoadMap;
 
+    private Map<ValueNumber, AvailableLoad> availableLoadReverseMap;
```

Comment:
I think it's better to introduce a class like the [BiMap from Guava](https://guava.dev/releases/19.0/api/docs/com/google/common/collect/BiMap.html)?
It will reduce the complexity of this class, and improve its readability.

We cannot use that `BiMap` because the value of `availableLoadReverseMap` is not an element but an array, but we can create a similar class on our own.

Reply:
Thanks a lot for looking into it, for this version I have applied the patch as-is.
I agree that the complexity of the class is high and using some kind of BiMap is a good idea.
I'll try implementing this differently